src/ansys_sphinx_theme/extension/linkcode.py

```diff
@@ -227,19 +227,18 @@ def link_code(app: Sphinx, doctree: Node):
 
             # Call user code to resolve the link
             try:
+                # Resolve the URI for source code link
                 uri = sphinx_linkcode_resolve(
                     domain=domain,
                     info=info,
                     library=library,
                     source_path=github_source,
                     github_version=github_version,
                 )
-                if not uri:
-                    # no source
-                    continue
 
-                if uri in uris or not uri:
-                    # only one link per name, please
+                # Check if the URI is already processed
+                if uri in uris:
```

Comment:
```suggestion
                if not uri or uri in uris:
```

Check that you have an object returned... and in case you have it, check that it is already in the uris list

If conditions are linear: you evaluate first the first statement, then the second statement... This will make it robust

Reply:
My life changed when I got to know about that behaviour of the `and` operator.